functions.php

```diff
@@ -215,15 +218,17 @@ function add_section_markup_before( $content, $section ) {
 		$style_attrs .= 'color: '. $text_color_custom .'; ';
 	}
 
+	$title = ! empty( $title ) ? ' data-section-link-title="' . $title . '"' : '';
+
 	ob_start();
 ?>
-	<section class="<?php echo $section_classes; ?>" style="<?php echo $style_attrs; ?>">
+	<section id="<?php echo $section->post_name; ?>" class="<?php echo $section_classes; ?>" style="<?php echo $style_attrs; ?>"<?php echo $title; ?>>
```

Comment:
I agree that we should auto-fill the ID value if possible.  Maybe the shortcode attribute could be a boolean--`disable_id="true"` instead?  Just to handle those one-off cases where we need to repeat sections that aren't necessarily worth creating an anchor for on the page.